Microsoft.NET.Build.Containers/ContainerHelpers.cs

```diff
@@ -9,6 +9,16 @@ public static class ContainerHelpers
 
     private static Regex imageNameRegex = new Regex("^[a-z0-9]+([._-][a-z0-9]+)*(/[a-z0-9]+([._-][a-z0-9]+)*)*$");
 
+    /// <summary>
+    /// Matches if the character is not lowercase or numeric
+    /// </summary>
+    private static Regex imageFirstNameCharacters = new Regex("[^a-z0-9]");
+    
+    /// <summary>
+    /// Matches if the string is not lowercase or numeric, or ., _, or -.
+    /// </summary>
+    private static Regex imageNameCharacters = new Regex("[^a-zA-Z0-9._-]");
```

Comment:
See above

Reply:
This one is used to match on a char-by-char basis and replace non-compliant chars with `-`.